bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/discovery/PlugwiseHADiscoveryService.java

```diff
@@ -0,0 +1,212 @@
+        if (localBridgeHandler != null) {
+            controller = localBridgeHandler.getController();
+        }
+
+        if (controller != null) {
+            DomainObjects domainObjects = controller.getDomainObjects();
+
+            if (domainObjects != null) {
+                for (Location location : domainObjects.getLocations().values()) {
+                    // Only add locations with at least 1 appliance (this ignores the 'root' (home)
+                    // location which is the parent of all other locations.)
+                    if (location.applianceCount() > 0) {
+                        locationDiscovery(location);
+                    }
+                }
+
+                for (Appliance appliance : domainObjects.getAppliances().values()) {
+                    // Only add appliances that are required/supported for this binding
+                    if (PlugwiseHABindingConstants.SUPPORTED_APPLIANCE_TYPES.contains(appliance.getType())) {
+                        applianceDiscovery(appliance);
+                    }
+                }
+            }
+        }
+    }
+
+    private void applianceDiscovery(Appliance appliance) {
+        String applianceId = appliance.getId();
+        String applianceName = appliance.getName();
+        String applianceType = appliance.getType();
+
+        PlugwiseHABridgeHandler localBridgeHandler = this.bridgeHandler;
+        if (localBridgeHandler != null) {
+            ThingUID bridgeUID = localBridgeHandler.getThing().getUID();
+
+            ThingUID uid;
+
+            Map<String, Object> configProperties = new HashMap<>();
+
+            configProperties.put(APPLIANCE_CONFIG_ID, applianceId);
+
+            switch (applianceType) {
+                case "thermostatic_radiator_valve":
+                    uid = new ThingUID(PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_VALVE, bridgeUID, applianceId);
+                    configProperties.put(APPLIANCE_CONFIG_LOWBATTERY, 15);
+                    break;
+                case "central_heating_pump":
+                    uid = new ThingUID(PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_PUMP, bridgeUID, applianceId);
+                    break;
+                case "heater_central":
+                    uid = new ThingUID(PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_BOILER, bridgeUID, applianceId);
+                    break;
+                case "zone_thermostat":
+                    uid = new ThingUID(PlugwiseHABindingConstants.THING_TYPE_APPLIANCE_THERMOSTAT, bridgeUID,
+                            applianceId);
+                    configProperties.put(APPLIANCE_CONFIG_LOWBATTERY, 15);
+                    break;
+                default:
+                    return;
+            }
+
+            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(uid).withBridge(bridgeUID)
+                    .withLabel(applianceName).withProperties(configProperties).withRepresentationProperty(applianceId)
```

Comment:
The representation property must be the name of the property, not its value.

Reply:
Sorry, fixed